tools/userland-mapping

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/python2.7
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2018 Adam Stevko
+#
+
+#
+# mapping.py - generate mapping between component FMRI and component package names, to be used for different purposes,
+# e.g. checking for outdated pacakges, vulnerable packages etc.
+#
+
+from __future__ import print_function, absolute_import
+
+import argparse
+import json
+import os
+import re
+import logging
+import subprocess
+import multiprocessing
+import sys
+
+try:
+    from scandir import walk
+except ImportError:
+    from os import walk
+
+logger = logging.getLogger('userland-mapping')
+
+COMONENT_MAPPING_FILENAME = 'mapping.json'
+
+
+def find_component_paths(path, subdir='components', debug=False):
+    expression = re.compile(r'.+\.p5m$', re.IGNORECASE)
+
+    paths = []
+    workspace_path = os.path.join(path, subdir)
+
+    for dirpath, dirnames, filenames in walk(workspace_path):
```

Comment:
Should we really walk the directory or better use git ls-files? What will happen when we walk non-clean oi-userland (with extracted archives and build directories)?

Reply:
Yes, we should as we are doing it every time when we are doing builds. That code is also in `bass-o-matic` and it isn't causing any issues :)